dotnet/src/Connectors/Connectors.AzureAIInference/Settings/AzureAIInferencePromptExecutionSettings.cs

```diff
@@ -42,7 +46,9 @@ public string? ExtraParameters
     /// decrease the likelihood of the model repeating the same statements verbatim.
     /// Supported range is [-2, 2].
     /// </summary>
-    public float? FrequencyPenalty
+    [JsonPropertyName("frequency_penalty")]
+    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
+    public double? FrequencyPenalty
```

Comment:
I wonder if the double precession is required for this and other double parameters?

Reply:
Every connnector is using `double`, I didn't want to do different for this one. I realized this on our UnitTests when setting the properties it was complaining that `0.7` is not a valid float only `0.7f` to make it work, which is no big deal but for the sake of pattern I kept the double here.